pkg/seachest/seachest.go

```diff
@@ -0,0 +1,156 @@
+		"Success",
+		"Failed",
+		"Not Supported",
+		"Cmd Failure",
+		"In-progress",
+		"Aborted",
+		"Bad Parameter",
+		"Memory Allocation Failed",
+		"Cmd Passthrough Failed",
+		"Library Mistmatch",
+		"Device Forzen",
+		"Permission Denied",
+		"File Open Error",
+		"Incomplete RFTRS",
+		"Cmd Time-out",
+		"Warning - Not all device enumerated",
+		"Invalid Checksum",
+		"Cmd not Available",
+		"Cmd Blocked",
+		"Cmd Interrupted",
+		"Unknown",
+         }
+         return seachestErrorSting[err]
+}
+
+// Identifier (devPath such as /dev/sda,etc) is an identifier for seachest probe
+type Identifier struct {
+	DevPath string
+}
+
+func (I *Identifier) SeachestBasicDiskInfo() (*C.driveInformationSAS_SATA, int) {
+
+	var device C.tDevice
+	var Drive C.driveInformationSAS_SATA
+	str := C.CString(I.DevPath)
+	defer C.free(unsafe.Pointer(str))
+
+	err := int(C.get_Device(str, &device))
+	if err != 0 {
+		glog.Error("Unable to get device info for device:%s with error:%s", I.DevPath, SeachestErrors(err))
+		return nil, err
+	}
+
+	err = int(C.get_SCSI_Drive_Information(&device, &Drive))
+	if err != 0 {
+		glog.Error("Unable to get derive info for device:%s with error:%s", I.DevPath, SeachestErrors(err))
+		return nil, err
+	}
+
+	//Added for cross verification, can be removed later on.
+	C.print_SAS_Sata_Device_Information(&Drive)
+
+	return &Drive, err
+}
+
+func (I *Identifier) GetHostName(driveInfo *C.driveInformationSAS_SATA) string {
+	return ""
+}
+
+func (I *Identifier) GetModelNumber(driveInfo *C.driveInformationSAS_SATA) string {
+	var ptr *C.char
+	ptr = &driveInfo.modelNumber[0]
+	str := C.GoString(ptr)
+	return str
+}
+
+func (I *Identifier) GetUuid(driveInfo *C.driveInformationSAS_SATA) string {
+	myString := fmt.Sprintf("%v", driveInfo.worldWideName)
+	return myString
+}
+
+func (I *Identifier) GetCapacity(driveInfo *C.driveInformationSAS_SATA) uint64 {
+	var capacity C.uint64_t
+	capacity = (C.uint64_t)(driveInfo.maxLBA * ((C.uint64_t)(driveInfo.logicalSectorSize)))
+	return ((uint64)(capacity))
+}
+
+func (I *Identifier) GetSerialNumber(driveInfo *C.driveInformationSAS_SATA) string {
+	var ptr *C.char
+	ptr = &driveInfo.serialNumber[0]
+	str := C.GoString(ptr)
+	return str
+}
+
+func (I *Identifier) GetVendorID(driveInfo *C.driveInformationSAS_SATA) string {
+	var ptr *C.char
+	ptr = &driveInfo.vendorID[0]
+	str := C.GoString(ptr)
+	return str
+}
+
+func (I *Identifier) GetPath(driveInfo *C.driveInformationSAS_SATA) string {
+	return I.DevPath
+}
+
+func (I *Identifier) GetFirmwareRevision(driveInfo *C.driveInformationSAS_SATA) string {
+	var ptr *C.char
+	ptr = &driveInfo.firmwareRevision[0]
+	str := C.GoString(ptr)
+	return str
+}
+
+func (I *Identifier) GetLogicalSectorSize(driveInfo *C.driveInformationSAS_SATA) uint32 {
+	return ((uint32)(driveInfo.logicalSectorSize))
+}
+
+func (I *Identifier) GetDiskType(driveInfo *C.driveInformationSAS_SATA) string {
+
+	if driveInfo.rotationRate == 0x0001 {
+		return "SSD"
+	}
+	return ""
```

Comment:
can we make this a bit better? SSD vs HDD perhaps instead of SSD vs ""?

Reply:
On second thought, if its *is* rotational, dont we not grab the RPM?

Reply:
We do see rotationRate = 0 in the wild. From SBC-3:

- 0 = not reported
- 1 = non-rotating (maybe SSD, maybe some array or other logical disk)
- 2 to 0x400 = reserved
- 0x401 to 0xfffe = rotation rate in RPM
- 0xffff = reserved

For some OSes, like linux, there is a /sys/block/DISK/queue/rotational flag that is set when RPM > 1. This is normally helpful enough.